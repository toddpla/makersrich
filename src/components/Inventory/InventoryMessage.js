import React, { Component } from 'react';

class InventoryMessage extends Component {

  handleMessage = () => {
    if(this.props.selected === undefined){
      return
    }
    if(this.props.selected.type === 'miscellaneous'){
      return console.log(this.props.selected);
      // return this.props.selected.properties[0].value
    }
    if (this.props.selected.properties && this.props.selected.properties[0].value) {
       return this.props.selected.properties[0].value
    } else {
      return `This is a ${this.props.selected.type}`
    }
  }

  render() {
    return (
      <div id="inventory-item-message">
        <h1>Information</h1>
        <p>{this.handleMessage()}</p>
      </div>
    );
  }

}

export default InventoryMessage;
