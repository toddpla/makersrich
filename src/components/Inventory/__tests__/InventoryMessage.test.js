import React from 'react';
import { shallow } from 'enzyme'
import InventoryMessage from '../InventoryMessage'

let wrapper

beforeEach(function() {
  wrapper = shallow(
    <InventoryMessage message={'message'}/>
  )
});

test('it renders InventoryMessage', () => {
  expect(wrapper).toMatchSnapshot()
})
