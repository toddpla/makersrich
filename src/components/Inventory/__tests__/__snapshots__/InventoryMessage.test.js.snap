// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders InventoryMessage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InventoryMessage
    selected="item"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Information
        </h1>,
        <div
          id="inventory-item-type"
        >
          <h2 />
        </div>,
        <div
          className="inventory-selected-item undefined"
          id="inventory-item-image"
        />,
        <div
          className="item-message"
        >
          This is a undefined
        </div>,
      ],
      "className": "inventory-item-message",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Information",
        },
        "ref": null,
        "rendered": "Information",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2 />,
          "id": "inventory-item-type",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "inventory-selected-item undefined",
          "id": "inventory-item-image",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is a undefined",
          "className": "item-message",
        },
        "ref": null,
        "rendered": "This is a undefined",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Information
          </h1>,
          <div
            id="inventory-item-type"
          >
            <h2 />
          </div>,
          <div
            className="inventory-selected-item undefined"
            id="inventory-item-image"
          />,
          <div
            className="item-message"
          >
            This is a undefined
          </div>,
        ],
        "className": "inventory-item-message",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Information",
          },
          "ref": null,
          "rendered": "Information",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2 />,
            "id": "inventory-item-type",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "inventory-selected-item undefined",
            "id": "inventory-item-image",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is a undefined",
            "className": "item-message",
          },
          "ref": null,
          "rendered": "This is a undefined",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
